rc/filetype/org.kak

```diff
@@ -0,0 +1,262 @@
+add-highlighter shared/org/inline/text/org_target       regex '[<]{2}[^\s<>][^\n<>]+[^\s<>][>]{2}' 0:link
+add-highlighter shared/org/inline/text/org_radio_target regex '[<]{3}[^\s<>][^\n<>]+[^\s<>][>]{3}' 0:link
+
+# Macros
+add-highlighter shared/org/inline/text/macros regex '[{]{3}[A-Za-z][a-zA-Z0-9-_]+\(.*?\)[}]{3}' 0:meta
+
+# Cookies
+add-highlighter shared/org/inline/text/cookies regex '[\[]\d+?(%|/\d+?)[\]]' 0:value
+
+# Commands
+# ‾‾‾‾‾‾‾‾
+
+define-command -hidden org-indent-on-new-line %{
+    evaluate-commands -draft -itersel %{
+        # follow list item indentation and amount of leading whitespaces
+        try %{ execute-keys -draft 'k<a-x>s\S<ret><space><a-?>^\h*([-+]|\h+[*]|\d[.)])<ret><a-x>s^\h*([+-]|\h+[*][^*]|\d[.)])\h*<ret>y/^\n<ret>PxHr<space>' }
+        try %{ execute-keys -draft -itersel 'k<a-x>s\h+$<ret>d' }
+    }
+}
+
+define-command -docstring "set up org TODO and PRIORITY items based on file contents" \
+org-parse-file %{ evaluate-commands -save-regs '"/' %{
+    # Parse TODO items
+    try %{
+        execute-keys -draft '/(?i)^\h*#\+(SEQ_|TYP_)?TODO:[^\n]+<ret><a-h>f:l<a-l>y: set-option buffer org_todo %reg{dquote}<ret>'
+        # This Perl code transforms these two lines to similar formats:
+        # `word1 word2 word3 word4' to `(word1|word2|word3)|(word4)',
+        # and `word1 word2 | word3 word4' to `(word1|word2)|(word3|word4)'.
+        # This regex is later used in to highlight todo items in headings.
+        set-option buffer org_todo %sh{ printf "%s\n" "${kak_opt_org_todo}" | perl -pe 'if (/^.*\|.*$/) {
+                                                                                             $_ =~ s/(.*)\|(.*)/($1)|($2)/;
+                                                                                             $_ =~ s/\(\s+/(/g;
+                                                                                             $_ =~ s/\s+\)/)/g;
+                                                                                             $_ =~ s/[\t ]+/|/g;
+                                                                                         } else {
+                                                                                             $_ =~ s/(.*) ([^\s]+$)/($1)|($2)/;
+                                                                                             $_ =~ s/\(\s+/(/g;
+                                                                                             $_ =~ s/\s+\)/)/g;
+                                                                                             $_ =~ s/[\t ]+/|/g;
+                                                                                         }' }
+    }
+    # Parse priority items
+    try %{
+        # this `execute-keys' selects first three letters in `#+PRIORITIES' line
+        execute-keys -draft '/(?i)^\h*#\+PRIORITIES:[^\n]+<ret><a-h>f:l<a-l>s\h*\w(\s+)?(\w)?(\s+)?(\w)?\s<ret>y: set-option buffer org_priority %reg{dquote}<ret>'
```

Comment:
Maybe this line would be clearer written as an `eval` with `exec`s in it.

Reply:
not sure what's in your mind

Reply:
Something like the following (untested), to make things more readable:

```
evaluate-commands -draft %{
    execute-keys /(?i)^\h*#\+PRIORITIES:[^\n]+<ret> \
                          <a-h>f:l<a-l> \
                          s\h*\w(\s+)?(\w)?(\s+)?(\w)?\s<ret> \
                          y
    set-option buffer org_priority %reg{dquote}
}
```

Reply:
Ah, got it, I'll redo this part